pkg/placement/workload-projector.go

```diff
@@ -364,41 +368,99 @@ func (wp *workloadProjector) newPerDestinationLocked(destination SinglePlacement
+type dynamicDuo struct {
+	apiVersion  string
+	namespaced  bool
+	preInformer upstreaminformers.GenericInformer // nil iff resource is namespaces
+	client      k8sdynamic.NamespaceableResourceInterface
+}
+
+func (wpd *wpPerDestination) getDynamicDuoLocked(gr metav1.GroupResource, apiVersion string, namespaced bool) (dynamicDuo, <-chan struct{}, error) {
 	if wpd.dynamicClient == nil {
 		mbwsName := SPMailboxWorkspaceName(wpd.destination)
 		mbwsCluster, have := wpd.wp.mbwsNameToCluster.Get(mbwsName)
 		if !have {
-			return nil, nil, errors.New("unable to map mailbox workspace name to cluster")
+			return dynamicDuo{}, nil, errors.New("unable to map mailbox workspace name to cluster")
 		}
 		wpd.dynamicClient = wpd.wp.dynamicClusterClient.Cluster(mbwsCluster.Path())
-		wpd.dynamicInformerFactory = k8sdynamicinformer.NewDynamicSharedInformerFactory(wpd.dynamicClient, 0)
+		justMine, err := labels.NewRequirement(ProjectedLabelKey, selection.Equals, []string{ProjectedLabelVal})
+		if err != nil {
+			panic(err)
+		}
+		justMineStr := justMine.String()
+		wpd.dynamicInformerFactory = k8sdynamicinformer.NewFilteredDynamicSharedInformerFactory(wpd.dynamicClient, 0,
+			metav1.NamespaceAll, func(opts *metav1.ListOptions) {
+				if opts.LabelSelector == "" {
+					opts.LabelSelector = justMineStr
+				} else {
+					opts.LabelSelector = opts.LabelSelector + "," + justMineStr
+				}
+			})
 		wpd.namespaceClient = wpd.wp.nsClusterClient.Cluster(mbwsCluster.Path())
 		wpd.namespacePreInformer = wpd.wp.nsClusterPreInformer.Cluster(mbwsCluster)
 		nsInformer := wpd.namespacePreInformer.Informer()
-		readyChan := make(chan struct{})
-		wpd.readyChan = readyChan
+		nsReadyChan := make(chan struct{})
+		wpd.nsReadyChan = nsReadyChan
 		go func() {
 			k8scache.WaitForNamedCacheSync("workload-projector", wpd.wp.ctx.Done(), nsInformer.HasSynced)
-			close(readyChan)
+			close(nsReadyChan)
 		}()
 		go nsInformer.Run(wpd.wp.ctx.Done())
 		wpd.dynamicInformerFactory.Start(wpd.wp.ctx.Done())
 	}
-	return wpd.dynamicClient, wpd.readyChan, nil
+	duo, have := wpd.preInformers.Get(gr)
+	if have {
+		if apiVersion != duo.apiVersion || namespaced != duo.namespaced {
+			wpd.logger.Error(nil, "Not implemented yet: changing version or namespaced of GroupResource", "groupResource", gr,
+				"oldVersion", duo.apiVersion, "newVersion", apiVersion,
+				"oldNamespaced", duo.namespaced, "newNamespaced", namespaced)
+			// TODO: implement
+		}
+	} else {
+		sgvr := MetaGroupResourceToSchema(gr).WithVersion(apiVersion)
+		wpd.logger.V(4).Info("Creating informer at destination", "groupResource", gr, "apiVersion", apiVersion, "namespaced", namespaced)
+		duo = dynamicDuo{
+			apiVersion: apiVersion,
+			namespaced: namespaced,
+			client:     wpd.dynamicClient.Resource(sgvr)}
+		if mgrIsNamespace(gr) {
+			// No way to know if a namespace is needed for other reasons,
+			// so no point in reacting to them.
```

Comment:
@waltforme : this is the comment I mentioned.

Reply:
Oh I see. I was testing the PR by running it, but not have enough bandwidth to review the code in full. Thanks for the info!